src/Accounts/Authentication.ResourceManager/ProtectedFileProvider.cs

```diff
@@ -87,18 +92,17 @@ public StreamWriter CreateWriter()
 
         protected virtual void InitializeStream()
         {
-            lock (_initializationLock)
+            _initializationLock.WaitOne();
```

Comment:
I believe we should use Mutex here, because `TryGetStreamLock()` may open files for write exclusively, see https://github.com/Azure/azure-powershell-common/blob/master/src/Authentication.Abstractions/DiskDataStore.cs#L324